openlibrary/catalog/marc/marc_base.py

```diff
@@ -30,11 +63,31 @@ def read_isbn(self, f):
             found.append(m.group(1))
         return found
 
-    def build_fields(self, want):
-        self.fields = {}
-        want = set(want)
-        for tag, line in self.read_fields(want):
-            self.fields.setdefault(tag, []).append(line)
+    def get_control(self, tag: str) -> str | None:
+        control = self.read_fields([tag])
+        _, v = next(control, (tag, None))
+        if tag == '008':
+            # Handle duplicate 008s, even though control fields are non-repeatable.
+            if others := [d for _, d in list(control) if len(d) == 40]:
```

Comment:
```suggestion
        if tag == '008' and isinstance(v, str):
            # Handle duplicate 008s, even though control fields are non-repeatable.
            if others := [d for _, d in list(control) if isinstance(d, str) and len(d) == 40]:
```

Reply:
Thanks @cclauss, that's helpful, and I think I can see a way forward now. I'd prefer is this were an `assert` at this point -- the `if` implies `v` might not be a `str` at this point, but really it will always be. _I_ know that, but I've been struggling to convince `mypy` :)

If it's not a string here, something is very wrong with the data, and there is likely code and protections which would have forced it to be a `str` already, or the data would have already been rejected.

Would appropriately placed `assert isinstance(v, str)` also satisfy mypy?

(I've just raised issue #7699 which is making it harder for me to check such things -- I thought I was missing something, but I think there's a problem with the Docker images.)

Reply:
Would appropriately placed assert isinstance(v, str) also satisfy mypy?

Yes.  It would.